crates/cxx-qt-gen/src/naming/rust.rs

```diff
@@ -257,10 +258,10 @@ mod tests {
     #[test]
     fn test_syn_type_cxx_bridge_to_qualified_mapped() {
         let mut mappings = TypeNames::default();
-        mappings.insert("A", Some(parse_quote! { ffi::B }), None, None);
+        mappings.insert("A", Some(format_ident!("ffi")), None, None);
```

Comment:
hmm I think the point of this test was to check that mapped names still get qualified.

eg

```rust
#[rust_name = "B"]
type A;
```

becomes `ffi::B` not `ffi:A` as it's been renamed. Is this possible to test still with the new mappings ?

Reply:
Not on the Rust side of things.

It was also not correct behavior before.

If I have this:
``` rust
mod ffi {
extern "Rust" {
    #[rust_name="B"]
    type A;
}
}
```

The type can only be referred to by `A`, not `B`.

Reply:
The correct identifier to use to refer to a type in CXX is always the `rust_name`.

Reply:
ah maybe it was this case

```rust
mod ffi {
  extern "C++" {
    #[rust_name="B"]
    type A;
  }
}
```

which becomes `ffi::B` to the outside and `B` inside the CXX bridge. `A` is never used ?

Reply:
With your example the insert call would have to be:
```
mappings.insert("B", Some(format_ident!("ffi")), Some("A"), None);
```
As in this case the `cxx_name` would be "A".

I don't know if that makes sense to test in the `rust`  part of the naming module?

Reply:
I just think if we had a test before that was checking something we should still test after, as it's easy to break things when refactoring. Whether the test is in this file or another doesn't matter. But we should ensure that having a cxx_name + the qualified method does the right thing

Reply:
Then in this case I think the right test would be:
```
mappings.insert("A", Some(format_ident!("ffi")), Some("B"), None);
```

As we want to test that a type that is mapped (i.e. has a `cxx_name` that is different from the `rust_name`) uses the right qualified access.

Will change to use that.